backend/SlothOrganizer/SlothOrganizer.Services/Tasks/TaskService.cs

```diff
@@ -40,5 +39,33 @@ public async Task<List<TaskDto>> Get(long dashboardId)
             var tasks = await _taskRepository.Get(dashboardId);
             return _mapper.Map<List<TaskDto>>(tasks);
         }
+
+        public async Task<TaskDto> Update(UpdateTaskDto updateTaskDto)
+        {
+            await _taskCompletionService.Update(updateTaskDto.TaskCompletion);
+            var task = _mapper.Map<UserTask>(updateTaskDto.Task);
+            var updatedTask = _mapper.Map<TaskDto>(await _taskRepository.Update(task));
+            if (updatedTask is null)
+            {
+                throw new EntityNotFoundException("Task with such id not found");
+            } 
+            var latestCompletion = updatedTask.TaskCompletions.MaxBy(tc => tc.End);
+            if (updateTaskDto.EndRepeating is DateTime endRepeating && updatedTask.TaskCompletions.Count > 1)
```

Comment:
All these if bodies, conditions could be separated into small functions.

Reply:
Fixed